### ISSUE TITLE
weights become 4x than usual 

### ISSUE THREAD

**Issue by BrianKenobi** [ultralytics/yolov5#2774]
## ❔Question
**### Update: I try to use coco128 and the origin model yolov5s and I got the same result as yolov5s weight is still 58.9 MB.**

Hello there, I'm training my custom data that only have 350 pictures.

But my yolov5 weights become very strange,
s weights is **58.9 MB**,
m weights is **172 MB**,
l weights is **378 MB.**

I have no idea how this happened.

I changed line 945 in datasets.py because my label is too long so the origin line make yolo can't detect my anchor : 

    return (w2 > wh_thr) & (h2 > wh_thr) & (w2 * h2 / (w1 * h1 + eps) > area_thr) & (ar < ar_thr)| ((ar < 240))

And I add this in front of the train.py because I don't have enough share memory:

```
import sys
import torch
from torch.utils.data import dataloader
from torch.multiprocessing import reductions
from multiprocessing.reduction import ForkingPickler

default_collate_func = dataloader.default_collate


def default_collate_override(batch):
  dataloader._use_shared_memory = False
  return default_collate_func(batch)

setattr(dataloader, 'default_collate', default_collate_override)

for t in torch._storage_classes:
  if sys.version_info[0] == 2:
    if t in ForkingPickler.dispatch:
        del ForkingPickler.dispatch[t]
  else:
    if t in ForkingPickler._extra_reducers:
        del ForkingPickler._extra_reducers[t]
```

I have changed batch-sizes and epoch from train.py and anchors from model yolov5l.yaml (didn't change anything down the backbone or head).
And I use 2 GPU to train so I use this command :

`python -m torch.distributed.launch --nproc_per_node 2 train.py --batch-size 32`

Even I git the newest code from github still have the same problem. 
Do you have any ideas or anything that I might accidentally change to make the weights became so big?

Thanks!

**Comment by glenn-jocher** [comment]
@BrianKenobi YOLOv5 training checkpoints should be 4X the size of final trained checkpoints, as they carry not just a FP16 model, but a FP16 [EMA](https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage) and an FP32 optimizer of the same size as the model (each model parameter has its own FP32 gradient saved within the optimizer). Final checkpoints contain only an FP16 model, with the EMA and optimizer both stripped after the final epoch of training:
https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/train.py#L435-L439

The `strip_optimizer()` function updates the checkpoint dictionary value for `model`, replacing it with `ema`, and sets `ema` and `optimizer` keys to `None`, which will reduce the checkpoint size by 3/4.
https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/utils/general.py#L495-L509

Final trained checkpoints after `strip_optimizer()` should match the README table sizes below.
<p align="center"><img width="700" alt="YOLOv5 Models" src="https://user-images.githubusercontent.com/26833433/103595982-ab986000-4eb1-11eb-8c57-4726261b0a88.png"></p>

**Comment by BrianKenobi** [comment]
I see!
Thank you @glenn-jocher .
